apps/emqx_rule_engine/src/emqx_rule_funcs.erl

```diff
@@ -785,6 +787,15 @@ is_map(_) -> false.
 is_array(T) when is_list(T) -> true;
 is_array(_) -> false.
 
+is_empty([]) ->
+    true;
+is_empty(<<>>) ->
+    true;
+is_empty(List) when is_list(List) ->
+    false;
+is_empty(Map) ->
+    ?MODULE:map_size(Map) == 0.
```

Comment:
Why use remote call (?MODULE:...)?

Reply:
map_size is a built-in function of Erlang. Here we need to call another function of the same name in this module.